kafka-statefulsets/scripts/kafka_run.sh

```diff
@@ -21,7 +21,7 @@ echo "LOG_DIR=$LOG_DIR"
 exec $KAFKA_HOME/bin/kafka-server-start.sh $KAFKA_HOME/config/server.properties \
 --override broker.id=$KAFKA_BROKER_ID \
 --override advertised.host.name=$(hostname -I) \
---override zookeeper.connect=$ZOOKEEPER_SERVICE_HOST:$ZOOKEEPER_SERVICE_PORT \
+--override zookeeper.connect=${KAFKA_ZOOKEEPER_CONNECT:-zookeeper:2181}  \
```

Comment:
I know that with current configuration, without `KAFKA_ZOOKEEPER_CONNECT` defined, this parameter fallback to zookeeper:2181. Does it make sense to use `$ZOOKEEPER_SERVICE_HOST:$ZOOKEEPER_SERVICE_PORT` instead of hard coded `zookeeper:2181` ? Finally what's the way for setting the `KAFKA_ZOOKEEPER_CONNECT` ? As parameter in the OpenShift template, or ?

Reply:
> Does it make sense to use $ZOOKEEPER_SERVICE_HOST:$ZOOKEEPER_SERVICE_PORT instead of hard coded zookeeper:2181?

I don't think that should be needed. We rely on DNS in other places as well (for example when building the Zookeeper cluster it self). So using the IP address from `$ZOOKEEPER_SERVICE_HOST` doesn't seem to have any benefits. And I would not expect the service port to change (no real need for it in OpenShift / Kubernetes), so I think we also don't need `$ZOOKEEPER_SERVICE_PORT`.

> Finally what's the way for setting the KAFKA_ZOOKEEPER_CONNECT ? As parameter in the OpenShift template, or ?

I don't think it makes sense to add it to the openShift template because it deployes Zookeeper always as zookeeper:2181. So making that configurable in the template would be useless. But if needed, it can be always passed to the Kafka pods as environment variable.

Reply:
> I don't think it makes sense to add it to the openShift template because it deployes Zookeeper always as zookeeper:2181

We could make that a parameter of the template. That said, I'm not sure it's worth doing right now.

Reply:
Well, we could. But unless you insist I would leave that to the cluster controller ;-)